src/main/java/io/hops/metadata/ndb/dalimpl/yarn/AppProvenanceClusterJ.java

```diff
@@ -0,0 +1,147 @@
+
+  @PersistenceCapable(table = TABLE_NAME)
+  public interface AppProvenanceEntryDto {
+
+    @PrimaryKey
+    @Column(name = ID)
+    String getId();
+
+    void setId(String id);
+
+    @Column(name = NAME)
+    String getName();
+
+    void setName(String name);
+    
+    @PrimaryKey
+    @Column(name = STATE)
+    String getState();
+
+    void setState(String state);
+
+    @Column(name = USER)
+    String getUser();
+
+    void setUser(String user);
+    
+    @PrimaryKey
+    @Column(name = TIMESTAMP)
+    long getTimestamp();
+
+    void setTimestamp(long timestamp);
+    
+    @PrimaryKey
+    @Column(name = SUBMIT_TIME)
+    long getSubmitTime();
+
+    void setSubmitTime(long submitTime);
+    
+    @PrimaryKey
+    @Column(name = START_TIME)
+    long getStartTime();
+
+    void setStartTime(long startTime);
+    
+    @PrimaryKey
+    @Column(name = FINISH_TIME)
+    long getFinishTime();
+
+    void setFinishTime(long finishTime);
+  }
+
+  @Override
+  public void addAll(Collection<AppProvenanceEntry> entries)
+    throws StorageException {
+    HopsSession session = connector.obtainSession();
+    ArrayList<AppProvenanceEntryDto> added = new ArrayList<>(entries.size());
+    try {
+      for (AppProvenanceEntry entry : entries) {
+        added.add(createPersistable(entry));
+      }
+      session.savePersistentAll(added);
+    } finally {
+      session.release(added);
+    }
+  }
+
+  @Override
+  public void add(AppProvenanceEntry entry) throws StorageException {
+    HopsSession session = connector.obtainSession();
+    AppProvenanceEntryDto dto = null;
+    try {
+      dto = createPersistable(entry);
+      session.savePersistent(dto);
+    } finally {
+      session.release(dto);
+    }
+  }
+
+  private AppProvenanceEntryDto createPersistable(AppProvenanceEntry entry) throws StorageException {
+    HopsSession session = connector.obtainSession();
+    AppProvenanceEntryDto dto = session.newInstance(AppProvenanceEntryDto.class);
+    dto.setId(entry.getId());
+    dto.setName(entry.getName());
+    dto.setState(entry.getState());
+    dto.setUser(entry.getUser());
+    dto.setTimestamp(entry.getTimestamp());
+    dto.setSubmitTime(entry.getSubmitTime());
+    dto.setStartTime(entry.getStartTime());
+    dto.setFinishTime(entry.getFinishTime());
+    return dto;
+  }
+
+  private Collection<AppProvenanceEntry> createCollection(Collection<AppProvenanceEntryDto> collection) {
+    ArrayList<AppProvenanceEntry> list = new ArrayList<>(collection.size());
+    for (AppProvenanceEntryDto dto : collection) {
+      list.add(createProvenanceLogEntry(dto));
+    }
+    return list;
+  }
+
+  private AppProvenanceEntry createProvenanceLogEntry(AppProvenanceEntryDto dto) {
+    return new AppProvenanceEntry(
+      dto.getId(),
+      dto.getName(),
+      dto.getState(),
+      dto.getUser(),
+      dto.getTimestamp(),
+      dto.getSubmitTime(), 
+      dto.getStartTime(),
+      dto.getFinishTime());
+  }
```

Comment:
Is there a plane to use these functions in the future? Write now they are not usefull as we only write to the database and never read from it.

Reply:
fixed